repos/system_upgrade/common/libraries/utils.py

```diff
@@ -176,3 +176,31 @@ def read_file(path):
     """
     with open(path, 'r') as f:
         return f.read()
+
+
+def require_exactly_one_message_of_type(model_class, error_class=StopActorExecutionError):
```

Comment:
I know this PR is probably still in draft state, but just FYI there is a discussion about similar functionality in framework here oamg/leapp#502, maybe it could be revived?